### ISSUE TITLE
1.3.0 can't find resources, but 1.2.1 can

### ISSUE THREAD

**Issue by paleite** [shakacode/sass-resources-loader#40]
Sass-resource-loader has been working fine with 1.2.1, but when I upgraded to 1.3.0 it stopped resolving the paths correctly.

Here's my error-output. Note that it is looking for something in `../../scss/` a folder that has never existed on my setup.

> ERROR in ./node_modules/css-loader!./node_modules/postcss-loader/lib!./node_modules/sass-loader/lib/loader.js!./node_modules/sass-resources-loader/lib/loader.js?{"resources":"[fullpath]/src/scss/_sass-resources.scss"}!./src/modules/header/header.scss
> Module build failed: 
> $header-height: 790px;
> ^
>       File to import not found or unreadable: ../../scss/src/scss/variables.

My `webpack.config.js`:
`  module: {
    rules: [
      {
        test: /\.scss$/,
        use: [
          'style-loader',
          'css-loader',
          'postcss-loader',
          'sass-loader',
          {
            loader: 'sass-resources-loader',
            options: {
              resources: path.resolve(__dirname, './src/scss/_sass-resources.scss')
            }
          }
        ]
      },
`

My `_sass-resources.scss`:

`@import '~bootstrap-sass/assets/stylesheets/bootstrap/variables';
@import '~bootstrap-sass/assets/stylesheets/bootstrap/mixins';
@import 'src/scss/variables';
@import 'src/scss/mixins';
`

Thanks in advance

**Comment by justin808** [comment]
@coditect and @alexfedoseev can you guys check this out? If not, I think I'll revert back to the older version.

**Comment by alexfedoseev** [comment]
I'll take a look. It might be an issue w/ `~` node modules prefix.

**Comment by coditect** [comment]
`@import` statements in Sass are relative to the file that contains them.  If `_sass-resources.scss` is in the same folder as `src/scss/variables` and `src/scss/mixins`, then those imports should really be: 
```
@import 'variables';
@import 'mixins';
```
The existing code worked prior to the addition of import rewriting because it relied on Sass using the project root as the base directory for resolving imports in SCSS files included via `sass-resources-loader`.

@alexfedoseev The `~` prefix should probably still to be looked at because there's nothing in `rewriteImports.js` to handle that syntax.

**Comment by paleite** [comment]
@coditect I'm afraid that didn't fix it either. It seems as though sass-resources-loader is trying to get the scss-files relative to its own location, rather than the project root. For example, doing ../../src/scss/variables worked for one scss-file that was 2 directories deep, but not for another file that was 3 directories deep.

**Comment by AymanJitan** [comment]
the latest build fails to load sass resources, here is an example
https://github.com/AymanJitan/webpackular/blob/master/webpack.config.js#L29
This used to work well in 1.2.1

**Comment by justin808** [comment]
@coditect @alexfedoseev @paleite @AymanJitan:   

Shall we rollback to 1.2.1 for 1.4 and create a beta version for 1.5 that includes this new fix from @coditect?

**Comment by coditect** [comment]
@justin808 Roll back as you see fit; I need to spend some time looking at compatibility with Webpack 1 (what @AymanJitan is using), Windows (see #41), and others.

**Comment by alexfedoseev** [comment]
@paleite @coditect Am I right that this issue is scoped to the use-case when scss is imported from `node_modules` and it doesn't affect common workflows? If it's so, I wouldn't rollback w/ version change, but patched current one and left a note in README w/ link to this issue. Please, confirm and I'll add a note.

Also, @coditect will you be able to look into this? I'm on the product feature atm and will be able to switch to this only after it's shipped (approx in 2-4 days).

**Comment by justin808** [comment]
@coditect @AymanJitan @paleite is the problem with the 1.3.0 release that it does not support Webpack v1? Maybe we can just promote 1.3 to v2, so it's clearly a major change?

**Comment by IAMtheIAM** [comment]
I think my [issue](https://github.com/shakacode/sass-resources-loader/issues/38) is related to this

**Comment by alexfedoseev** [comment]
@coditect @AymanJitan @paleite @IAMtheIAM @justin808 I will merge https://github.com/shakacode/sass-resources-loader/pull/45 once it's ready. If this PR won't fix these issues, I incline to remove support for SCSS imports, b/c we don't use it and there are no maintainers to take care of it. No feature is better than a buggy one.

**Comment by alexfedoseev** [comment]
Also, please comment on https://github.com/shakacode/sass-resources-loader/issues/46

**Comment by imakatman** [comment]
Any update on this issue?

**Comment by IAMtheIAM** [comment]
Please fix and upload next version to npm, its really a pain to have to downgrade back to 1.2.1 every time I run npm-check-updates

**Comment by justin808** [comment]
@imakatman, @IAMtheIAM I'm happy to consider a PR. 

@alexfedoseev is this fixed but not yet released?

**Comment by alexfedoseev** [comment]
`1.3.1` is released. Feel free to reopen if it won't solve this issue.

**Comment by paleite** [comment]
The problem still persists, unfortunately.

**Comment by TMiller00** [comment]
The problem persists.

This is the setup I'd I like to achieve. 
```
options: {
    resources: [                                        
        './node_modules/bourbon/app/assets/stylesheets/**/._*.scss',
        './node_modules/bourbon-neat/app/assets/stylesheets/**/._*.scss',
        './assets/styles/_custom.scss',
    ],
}
```

I receive two separate errors under different circumstances

1) When I build with the above setup, the build fails and I get an error saying that the mixins that should be in `bourbon-neat` can't be found
2) When I comment out the resource for `_custom.scss`, I get the error saying `Something wrong with provided resources. Make sure 'options.resources' is String or Array of Strings.`

**Comment by chetzof** [comment]
Can confirm

**Comment by paleite** [comment]
Any update on this? The problem still persists here.

@justin808 : This is related to Webpack v2/3 in my case.

@alexfedoseev : No, not node_modules. It's trying to import from the src, but from a path that's never existed before. It's resolving incorrectly. in > 1.2.1

**Comment by alexfedoseev** [comment]
@paleite I'm going to revert changes introduced in `1.3.0` tomorrow.

**Comment by IAMtheIAM** [comment]
I'm using 1.3.1 and it resolves the files just fine. Did you revert the changes in 1.3.1 or in the current unreleased version @alexfedoseev ?

**Comment by alexfedoseev** [comment]
@IAMtheIAM I haven't done anything yet (since `1.3.1` release).

**Comment by paleite** [comment]
Here's a repo with the bug reproduced:
https://github.com/paleite/sass-resources-loader-bug

Instructions included. Hope this helps!

Ping @alexeyMohnatkin @justin808 @coditect

**Comment by alexfedoseev** [comment]
@paleite Replace content of your resources file w/ this:

```scss
@import "~bootstrap-sass/assets/stylesheets/bootstrap/variables";
@import "~bootstrap-sass/assets/stylesheets/bootstrap/mixins";
@import "./variables";
```

**Comment by alexfedoseev** [comment]
See added notes: https://github.com/shakacode/sass-resources-loader/commit/f3d2cf6245abac5cc7e03799ab2a810d84a61b91

**Comment by paleite** [comment]
@alexfedoseev Works wonderfully!
